sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Provides the client options for serializing models.
+    /// </summary>
+    public class ModelSerializerOptions
```

Comment:
Do we provide anything that tells the serialize how to handle cycles in the object graph?  If the object graph has cycles, what happens in Serialize?

Reply:
This is completely left up to the implementation of serialization.  The public surface purposefully has very little control over how things get serialized because the goal is not to create a generic serialization story but to create public access to our already implemented internal serialization code.

Reply:
I agree that it should be left to the serializer, but do we have enough for our serializers to implement cycle detection (and throw). The main thing I worry about is that serializers need state to track cycles and I am not sure where such state would be stored given we simply pass Utf8Writer around.

Reply:
I prefer if we open a new issue on this and track it separately as we don't have this today and haven't needed it so far.  I agree its probably a good problem to solve but out of scope for this PR.

Reply:
https://github.com/Azure/autorest.csharp/issues/3657